### ISSUE TITLE
Missing symbols in official Windows TensorFlow native builds

### ISSUE THREAD

**Issue by karllessard** [tensorflow/tensorflow#62579]
Hi, we (SIG-JVM) are trying to leverage the Windows C Library builds available under https://storage.googleapis.com/tensorflow/libtensorflow/ which, I think, are built by Google. 

Starting from TF 2.14.x, these builds are broken because they do not export the TSL symbols required by the TensorFlow C API.

Here is the list of missing symbols that we (TF Java) do depend on:

```
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_DeleteStatus
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_SetStatus
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_SetStatusFromIOError
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_Status
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_Message
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_ForEachPayload
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_NewStatus
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_GetCode
jnitensorflow.obj : error LNK2001: unresolved external symbol TSL_SetPayload
```

Can someone look at this please? (modifié)

**Comment by angerson** [comment]
It seems like Intel has fixed something related that will land in 2.16: https://github.com/tensorflow/tensorflow/issues/61830

Does that help?

**Comment by karllessard** [comment]
Is there any Windows build available in the [libtensorflow-nightly](https://storage.googleapis.com/libtensorflow-nightly) bucket?

**Comment by nitins17** [comment]
> Is there any Windows build available in the [libtensorflow-nightly](https://storage.googleapis.com/libtensorflow-nightly) bucket?

The latest nightly libtensorflow archive for Windows can be found [here](https://storage.googleapis.com/libtensorflow-nightly/prod/tensorflow/release/windows/latest/cpu/windows_cpu_libtensorflow_binaries.tar.gz).

**Comment by karllessard** [comment]
Thanks @nitins17 , we just checked and while the nightly builds do have some of the TSL headers, the `TSL_` symbols are still missing from the binaries. cc\ @angerson

**Comment by karllessard** [comment]
Any updates about this/any PR coming soon? Thanks!

**Comment by mraunak** [comment]
Hi @karllessard, PR is in progress, will have it merged to master within a couple of days.

**Comment by karllessard** [comment]
Adding a reference to this comment here as well, problem is still not fixed after merging #62874: https://github.com/tensorflow/tensorflow/pull/62874#issuecomment-1926007848

**Comment by nitins17** [comment]
Can you provide a short reproducible code snippet for this?

**Comment by karllessard** [comment]
I can work on a something yes, though I don't have myself a Windows machine and it may take me some time. Meanwhile, you can look at our CI/CD where our own build (TF Java) is failing: https://github.com/tensorflow/java/actions/runs/7777626622/job/21206146416#step:6:1655

All other platforms (linux, macos) succeed to link successfully to the C lib, including to TSL symbols.

**Comment by karllessard** [comment]
Just to highlight here that it would be great to have this fix ready for the 2.16 cut, so that TF Java can continue to support Windows properly in its forthcoming  next release.

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The PR raised for the similar issue was aslo merged. https://github.com/tensorflow/java/pull/518

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
